mistakedetection.tests/src/ca/mcgill/sel/mistakedetection/tests/MistakeDetectionPerformanceAnalysis.java

```diff
@@ -264,6 +342,298 @@ public void testStudentSolution20() {
+
+    studentData.put(String.valueOf(nId++), new Object[] {""});
+    studentData.put(String.valueOf(nId++), new Object[] {"", "", "Actual Vs MDS"});
+
+    studentData.put(String.valueOf(nId++),
+        new Object[] {"Correct Identification % (mistakes)", "", "=C" + tpIndex + "/E" + totalMistakeindex + "*100"});
+    studentData.put(String.valueOf(nId++), new Object[] {"Correct Identification % (verdicts)", "",
+        "=((B" + totalNumbVerdictsIndex + "-F" + MDSvsActualIndex + ")/B" + totalNumbVerdictsIndex + ")*100"});
+    studentData.put(String.valueOf(nId++), new Object[] {"True Negative", "", "=SUM(IF(D2:D" + instructorElems
+        + "=0, IF(E2:E" + instructorElems + "=0, 1,0), 0 )) + (102 - B" + instElemsIndex + ")"});
+    studentData.put(String.valueOf(nId++), new Object[] {"True Positive", "",
+        "=SUM(IF(D2:D" + id + ">0, IF(D2:D" + id + "> E2:E" + id + ", E2:E" + id + ", D2:D" + id + "), 0 ))"});
+    studentData.put(String.valueOf(nId++), new Object[] {"False Positive", "",
+        "=SUM(IF(E2:E" + id + "> D2:D" + id + ", E2:E" + id + " -D2:D" + id + "), 0 )+B" + addFalsePosIndex});
+    studentData.put(String.valueOf(nId++), new Object[] {"False Negative", "",
+        "=SUM(IF(D2:D" + id + ">E2:E" + id + ", D2:D" + id + " - E2:E" + id + "), 0 )+B" + addFalseNegIndex});
+    studentData.put(String.valueOf(nId++),
+        new Object[] {"Recall (TP / (TP + FN))", "", "=(C" + tpIndex + "/(C" + tpIndex + "+C" + fnIndex + "))"});
+    studentData.put(String.valueOf(nId++),
+        new Object[] {"Precision (TP / (TP + FP))", "", "=(C" + tpIndex + "/(C" + tpIndex + "+C" + fpIndex + "))"});
+    
+    studentData.put(String.valueOf(nId++),
+        new Object[] {"F1 (2 * Precision * Recall / (Precision + Recall))", "", "=(2*C"+precisionIndex+"*C"+recallIndex+")/(C"+precisionIndex+"+C"+recallIndex+")"});
+    studentData.put(String.valueOf(nId++),
+        new Object[] {"F2 (5 * Precision * Recall) / (4 * Precision + Recall)", "", "=(5*C"+precisionIndex+"*C"+recallIndex+")/(4*C"+precisionIndex+"+C"+recallIndex+")"});
+    
+
+    Set<String> keyid = studentData.keySet();
+    int rowid = 0;
+
+    for (String key : keyid) {
+      row = spreadsheet.createRow(rowid++);
+      Object[] objectArr = studentData.get(key);
+      int cellid = 0;
+      for (Object obj : objectArr) {
+        Cell cell = row.createCell(cellid++);
+        if (obj instanceof String) {
+          String sObj = (String) obj;
+          if (sObj.startsWith("=")) {
+            sObj = sObj.replaceFirst("=", "");
+            cell.setCellFormula(sObj);
+          } else {
+            cell.setCellValue(sObj);
+          }
+        } else {
+          cell.setCellValue((int) obj);
+        }
+      }
+    }
+    FileOutputStream out = new FileOutputStream(new File(OUTPUT_LOC + "GroundTruth_" + name + ".xlsx"));
+
+    workbook.write(out);
+    out.close();
```

Comment:
Is it possible to automatically close the output stream using `AutoClosable`?

Reply:
Is the current approach problematic?

Reply:
If there is an exception in the middle of the function execution, the file will not be closed properly which is a problem on certain filesystems, so a `try` is needed here